components/action-mgt/org.wso2.carbon.identity.action.execution/src/main/java/org/wso2/carbon/identity/action/execution/ActionExecutorServiceImpl.java

```diff
@@ -466,22 +591,47 @@ private List<PerformableOperation> validatePerformableOperations(ActionExecution
                 .collect(Collectors.toList());
 
-        if (LOG.isDebugEnabled()) {
             // todo: add to diagnostics
-            List<String> allowedOps = new ArrayList<>();
-            List<String> notAllowedOps = new ArrayList<>();
-
-            response.getOperations().forEach(operation -> {
-                String operationDetails = "Operation: " + operation.getOp() + " with path: " + operation.getPath();
-                if (allowedPerformableOperations.contains(operation)) {
-                    allowedOps.add(operationDetails);
-                } else {
-                    notAllowedOps.add(operationDetails);
+            if (LOG.isDebugEnabled() || LoggerUtils.isDiagnosticLogsEnabled()) {
+                List<String> allowedOps = new ArrayList<>();
+                List<String> notAllowedOps = new ArrayList<>();
+
+                response.getOperations().forEach(operation -> {
+                    String operationDetails = "Operation: " + operation.getOp() + " with path: " + operation.getPath();
+                    if (allowedPerformableOperations.contains(operation)) {
+                        allowedOps.add(operationDetails);
+                    } else {
+                        notAllowedOps.add(operationDetails);
+                    }
+                });
+                LOG.debug("Allowed Operations: " + String.join(", ", allowedOps) +
+                        ". Not Allowed Operations: " + String.join(", ", notAllowedOps));
+                if (LoggerUtils.isDiagnosticLogsEnabled()) {
+                    DiagnosticLog.DiagnosticLogBuilder diagLogBuilder = new DiagnosticLog.DiagnosticLogBuilder(
+                            ActionExecutionConstants.LogConstants.ACTION_EXECUTION,
+                            ActionExecutionConstants.LogConstants.ActionIDs.VALIDATE_ACTION_OPERATIONS);
+                    diagLogBuilder
+                            .configParam("action id", action.getId())
+                            .configParam("action type", action.getType().getActionType())
+                            .configParam("action endpoint", action.getEndpoint().getUri())
+                            .configParam("action endpoint authentication type",
+                                    action.getEndpoint().getAuthentication().getType().getName())
+                            .configParam("allowed operations", allowedOps.isEmpty() ? "empty" : allowedOps)
+                            .configParam("not allowed operations", notAllowedOps.isEmpty() ? "empty" : notAllowedOps)
+                            .resultMessage(
+                                    "Validated operations to perform on " + action.getType().getActionType()
+                                            + " action.")
+                            .logDetailLevel(DiagnosticLog.LogDetailLevel.APPLICATION)
+                            .resultStatus(DiagnosticLog.ResultStatus.SUCCESS)
+                            .build();
```

Comment:
Can you post the log entry in the file here

Reply:
Updated in the PR description